docs/scos/dev/front-end-development/202307.0/marketplace/ui-components-library/datasources/datasource-trigger/datasource-trigger.md

```diff
@@ -0,0 +1,140 @@
+---
+title: Datasource Trigger
+description: Details about the Datasource Trigger service in the components library.
+template: concept-topic-template
+related:
+  - title: Datasource Dependable
+    link: docs/scos/dev/front-end-development/page.version/marketplace/ui-components-library/datasources/datasource-dependable.html
+  - title: Datasource Http
+    link: docs/scos/dev/front-end-development/page.version/marketplace/ui-components-library/datasources/datasource-http.html
+  - title: Datasource Inline Table
+    link: docs/scos/dev/front-end-development/page.version/marketplace/ui-components-library/datasources/datasource-inline-table.html
+  - title: Datasource Inline
+    link: docs/scos/dev/front-end-development/page.version/marketplace/ui-components-library/datasources/datasource-inline.html
+---
+
+This document describes the Datasource Trigger service in the components library.
+
+## Overview
+
+Datasource Trigger is an Angular service that provides a flexible way to fetch data based on user-triggered events.
+
+```html
+<spy-select
+    [datasource]="{
+        type: 'trigger',
+        event: 'change',
+        datasource: {
+            type: 'http',
+            url: '/request-url',
+        },
+    }"
+>
+</spy-select>
+```
+
+The trigger element, `SelectComponent` in the example, must implement a `DatasourceTriggerElement` abstract class (token) and return a component instance using the `getTriggerElement()` abstract method:   
+
+```ts
+@Component({
+    ...,
+    providers: [
+        {
+            provide: DatasourceTriggerElement,
+            useExisting: SelectComponent,
+        },
+    ],
+})
+export class SelectComponent implements DatasourceTriggerElement {
+    ...
+    getTriggerElement(): Observable<HTMLElement> {
+        // This method must return an Observable of the component instance.
+    }
+    ...
+}
+```
+
+## Main service
+
+The main module lets you register any `datasource.trigger` by key using the `withEvents()` static method. It assigns the object of datasources to `DatasourceEventTypesToken` under the hood.
+
+The `resolve()` method gets the trigger element using the `DatasourceTriggerElement` abstract class (token), finds a specific service from `DatasourceEventTypesToken` by `config.event` (from the argument) and returns observable with data (based on the data returned from trigger element) by `Datasource.resolve()`.
```

Comment:
Split this into several sentences and try to remove the braces

Reply:
In this case context will be missed